### ISSUE TITLE
[consistent-type-imports] restrict report to the specific offending

### ISSUE THREAD

**Issue by Josh-Cena** [typescript-eslint/typescript-eslint#4915]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/consistent-type-imports": "error"
  }
}
```

```TS
import { a, type b } from "c";

type c = a;
type d = b;
```

[playground](https://typescript-eslint.io/play#ts=4.6.2&sourceType=module&code=JYWwDg9gTgLgBAbzgQwDRxgTzAUzgIzgF84AzKCEOAIgGNqBuAKCa1zlrgF4Vm28AJtwLMgA&rules=N4IgAgLgngDgpgZwMYCcCWMIFpEBs0B2EA9EgPYEJoIRxFbTxZoC2MZKECIAXCHChQcQAXyA&tsConfig=N4XyA)

**Expected Result**

The rule only reports the first `ImportSpecifier`

**Actual Result**

The rule reports the entire `ImportDeclaration`, which makes debugging a little harder when the declaration is long.

I'm not using autofix, because #4338 is not there yet.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.22.0` |
| `@typescript-eslint/parser`        | `5.22.0` |
| `TypeScript`                       | `4,6.2` |
| `ESLint`                           | `8.14.0` |
| `node`                             | `18.0.0` |

**Comment by Josh-Cena** [comment]
The fifteen levels of indentation in that rule's source makes me want to instantly quit coding every time I look at it

**Comment by Josh-Cena** [comment]
The fifteen levels of indentation in that rule's source makes me want to instantly quit coding every time I look at it

**Comment by bradzacher** [comment]
Yeah the fixer is dog ugly logic to implement unfortunately because imports are so damn complex.

Most of the rule logic isn't THAT bad though. It probably needs a pass over it to rewrite it now that it's written and finalised.

**Comment by kirkwaiblinger** [comment]
I think that this kind of implies having a separate loc for each reported import, too, for example 
```
import { a, type b, c } from "outside-module";
         ^          ^

type A = a;
type B = b;
type C = c;
```
This is kinda of problematic, until https://github.com/typescript-eslint/typescript-eslint/issues/8554 is closed (which looks to be very soon!), due to the amount of fixer logic in the rule. With multipass fixes, I think we can just have each symbol report and fix "individually" but still assert that the actual result is correct

**Comment by bradzacher** [comment]
TBH idk if we should do individual reporting if there are multiple per import.
The fixer logic is complicated AF and refactoring it to be able to do each specifier in isolation will be 🤮

**Comment by bradzacher** [comment]
@Josh-Cena the issue with multiple report locations is that it creates multiple fixers and those fixers cannot talk to one another - so either they will conflict or they'll do the same thing

For example imagine this
```ts
import { a, b, c } from "outside-module";
         ^     ^
```

If we want to fix this to an inline style it's easy - for sure:
```diff
// fix 1
-import { a, b, c } from "outside-module";
+import { type a, b, c } from "outside-module";

// fix 2
-import { a, b, c } from "outside-module";
+import { a, b, type c } from "outside-module";
```

These fixes are non-overlapping so easy as.
But if you want to fix to a top-level style:
```
// fix 1
-import { a, b, c } from "outside-module";
+import { b, c } from "outside-module";
+import type { a } from "outside-module";

// fix 2
-import { a, b, c } from "outside-module";
+import { a, b } from "outside-module";
+import type { c } from "outside-module";
```

Now you have this issue either
- eslint will see that the fixers add the same line and so it'll reject the second fixer - _which will guarantee an extra lint pass to recalculate and apply the second fix,_ or
- eslint will see two non-overlapping insertions - causing two import statements to be inserted for the same module

Either way - that's not a great result! A slower lint run or duplicate code. IMO that's not a great trade-off for the end user for what is a relatively minor improvement.

**Comment by kirkwaiblinger** [comment]
Has the idea of a non-contiguous loc for a single report been thought about before? That would allow us to get semantically sensible underlines but one fix... Would also make sense for `method-signature-style` where multiple method declarations, currently flagged separately, get collapsed to 1 by the fix

**Comment by bradzacher** [comment]
I don't think it has - it's a pretty fundamental change to how eslint does things.

I think there's huge value in it eg rust's clippy does multi range reports (but only for "related spans", not primary ones) and it's a great way to expose additional info to users.

**Comment by Josh-Cena** [comment]
What about this: we still uphold the principle of "report what will be changed".

- If we need to split out a separate statement, report the statement as a whole.
- If we are changing a few identifiers (inline style), report those identifiers.
